public/js/sheets.js

```diff
@@ -0,0 +1,63 @@
+async function loadCountyHouseholdMap(){
+  return new Promise((resolve, reject) => {
+    Papa.parse(
+      'https://docs.google.com/spreadsheets/d/1jNk90L1FGXt3estVzFII2-eeKQZ85RYiLKCyNe14nGg/export?format=csv&gid=679979845',
+      {
+        download: true,
+        header: true,
+        complete: function (results) {
+          const countyHouseholdMap = results.data;
+          setLocalItem('countyHouseholdMap', countyHouseholdMap);
+          resolve(countyHouseholdMap)
+        },
+        error (err) {
+          reject(err)
+        }
+      }
+    );
+  })
+}
+
+
+
+async function init() {
+  if (!getLocalItem('countyHouseholdMap')) {
+    loadCountyHouseholdMap();
+  }
+}
+
+
+function getLocalItem(key) {
+  const localItem =  JSON.parse(localStorage.getItem(key));
+  if(!localItem || !localItem.expiry || Date.now>localItem.expiry){
+      return null
+  }
+  return localItem.data;
+}
+
+
+function setLocalItem(key, item) {
+  return localStorage.setItem(
+    key,
+    JSON.stringify({ data: item, expiry: Date.now() + 3 * 60 * 100 })//expiry = 3 mins
+  ); 
+}
+
+async function gethouseHold(county) {
+  let countyHouseholdMap = getLocalItem('countyHouseholdMap');
+  if (!countyHouseholdMap) {
+    await loadCountyHouseholdMap();
+    countyHouseholdMap = getLocalItem('countyHouseholdMap');
+  }
+  const householdInfo = countyHouseholdMap && countyHouseholdMap.find(
+    (row) => row.County.toLowerCase().trim() === county.toLowerCase().trim()
+  );
+  if (!householdInfo) {
+    return 0;
+  }
+  return Number(householdInfo.Households);
+}
+
+
+
+window.addEventListener('DOMContentLoaded', init);
```

Comment:
`init is` an _`async`_ function. Does it play nice with event listeners?

Reply:
yes, it does. I can omit the async, but the `loadCountyHouseholdMap` is async anyway. this is to ensure the function is loaded on startup.

Reply:
As long as you're sure it works, I'm good for it.